src/modules/applepay/applepay.test.ts

```diff
@@ -0,0 +1,59 @@
+import { describe, expect, it } from "vitest";
+import { validateMerchant } from "./applepay";
+import { setupRecording } from "@/__tests__/polly";
+import { testEnv } from "@/__tests__/test-env.mjs";
+describe(`Apple Pay`, () => {
+  setupRecording({});
+
+  it(`should create Apple Merchant Session`, async () => {
+    await expect(
+      validateMerchant({
+        domain: "saleor.tofw.pl",
+        merchantIdentifier: "merchant.com.adyen.saleorMichal.test",
+        merchantName: "Michal Miszczyszyn Test Merchant ID",
+        validationURL: "https://apple-pay-gateway.apple.com/paymentservices/paymentSession",
+        applePayCertificate: testEnv.TEST_ADYEN_APPLEPAY_CERTIFICATE,
+      }),
+    ).resolves.toEqual(expect.any(Object));
+  });
+
+  it(`should fail creating Apple Merchant Session - invalid domain`, async () => {
+    await expect(
+      validateMerchant({
+        domain: "saleor.tofw.com",
+        merchantIdentifier: "merchant.com.adyen.saleorMichal.test",
+        merchantName: "Michal Miszczyszyn Test Merchant ID",
+        validationURL: "https://apple-pay-gateway.apple.com/paymentservices/paymentSession",
+        applePayCertificate: testEnv.TEST_ADYEN_APPLEPAY_CERTIFICATE,
+      }),
+    ).rejects.toThrowErrorMatchingInlineSnapshot(
+      '"{\\"statusCode\\":\\"400\\",\\"statusMessage\\":\\"Payment Services Exception merchantId=482F45B7C387B70E52CF8F2C2E240E7B198C18FF3BCE2284A3FC617BDBFA16D1 not registered for domain=saleor.tofw.com\\"}"',
+    );
+  });
+
+  it(`should fail creating Apple Merchant Session - invalid merchant identifier`, async () => {
+    await expect(
+      validateMerchant({
+        domain: "saleor.tofw.pl",
+        merchantIdentifier: "merchant.com.adyen.doesntexistlol.test",
+        merchantName: "Michal Miszczyszyn Test Merchant ID",
+        validationURL: "https://apple-pay-gateway.apple.com/paymentservices/paymentSession",
+        applePayCertificate: testEnv.TEST_ADYEN_APPLEPAY_CERTIFICATE,
+      }),
+    ).rejects.toThrowErrorMatchingInlineSnapshot(
+      '"{\\"statusCode\\":\\"417\\",\\"statusMessage\\":\\"Payment Services Exception merchantId=482F45B7C387B70E52CF8F2C2E240E7B198C18FF3BCE2284A3FC617BDBFA16D1 unauthorized to process transactions on behalf of merchantId=01671331980C1600D0E8910C6EDC28552303C240245B920B3AF531E767F47184 reason=01671331980C1600D0E8910C6EDC28552303C240245B920B3AF531E767F47184 is not a registered merchant in WWDR and isn\'t properly authorized via Mass Enablement, either.\\"}"',
+    );
+  });
+
+  it(`should fail creating Apple Merchant Session - invalid URL`, async () => {
+    await expect(
+      validateMerchant({
+        domain: "saleor.tofw.pl",
+        merchantIdentifier: "merchant.com.adyen.saleorMichal.test",
+        merchantName: "Michal Miszczyszyn Test Merchant ID",
+        validationURL: "https://apple-pay-gateway.apple.com/paymentservices/paymentSession2",
```

Comment:
We could make this more obvious we're checking the `validationURL` in this test:
```suggestion
        validationURL: "https://invalid-validation-url.com",
```

Reply:
No because changing the domain will result in another error being thrown. I'll add a test for that too.